python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -1365,6 +1365,92 @@ def test_from_torch_e2e(ray_start_regular_shared, enable_optimizer, tmp_path):
     _check_usage_record(["FromItems"])
 
 
+def test_limit_pushdown(ray_start_regular_shared, enable_optimizer):
+    def f1(x):
+        return x
+
+    def f2(x):
+        return x
+
+    # Test basic limit pushdown past Map, which happens on the logical plan.
+    ds = ray.data.range(100, parallelism=100).map(f1).limit(1).materialize()
+    assert (
+        str(ds._plan._logical_plan.dag)
+        == "Read[ReadRange] -> Limit[Limit] -> MapRows[Map(f1)]"
+    )
+    assert len(ds.take_all()) == 1
+
+    # Test basic Limit -> Limit fusion, which applies
+    # during physical plan optimization.
+    ds2 = ray.data.range(100).limit(5).limit(100).materialize()
+    assert (
+        str(ds2._plan._logical_plan.dag)
+        == "Read[ReadRange] -> Limit[Limit] -> Limit[Limit]"
```

Comment:
I think we can just fuse the limit ops in the logical plan. 
And could you also update the Limit op name to include the limit number?